src/main/java/com/google/sps/servlets/utils/Statistics.java

```diff
@@ -15,93 +15,140 @@
+  private static final BigDecimal LOWER_SCORE = BigDecimal.valueOf(LOWER_SCORE_VAL);
+  private static final Comparator<UserComment> ascendingScoreCompare =
+      (UserComment o1, UserComment o2) -> Double.compare(o1.getScore(), o2.getScore());
 
-  // Contains sentiment scores in the range [-1, 1] with given intervals.
-  private Map<Range, Integer> aggregateValues;
+  // Contains sentiment bucket information for all intervals
+  private List<SentimentBucket> sentimentBucketList;
+  private double averageMagnitude;
   private double averageScore;
 
-  public Map<Range, Integer> getAggregateValues() {
-    return aggregateValues;
+  public List<SentimentBucket> getSentimentBucketList() {
+    return sentimentBucketList;
+  }
+
+  public double getAverageMagnitude() {
+    return averageMagnitude;
   }
 
   public double getAverageScore() {
     return averageScore;
   }
 
   /**
-   * Constructor of Statistics to filter out invalid sentiment scores, set aggregate hash map and
-   * average score.
+   * Constructor of Statistics to get average score and magnitude and create aggregate sentiment
+   * bucket list to store each interval's information
    *
-   * @param sentimentScores given score values
+   * @param userCommentList given list of userComment objects
+   * @param topNComments the number of highest magnitudes to retrieve
    */
-  public Statistics(List<Double> sentimentScores) throws RuntimeException {
-    sentimentScores =
-        sentimentScores.stream()
-            .filter(score -> (score >= LOWER_END_VAL && score <= UPPER_END_VAL))
-            .collect(Collectors.toList());
-    setAggregateScores(sentimentScores);
-    setAverageScore(sentimentScores);
+  public Statistics(List<UserComment> userCommentList, int topNComments) {
+    sentimentBucketList =
+        categorizeToBucketList(
+            userCommentList, LOWER_SCORE, UPPER_SCORE, SCORE_INTERVAL, topNComments);
+    averageScore = getAverageScore(userCommentList);
+    averageMagnitude = getAverageMagnitude(userCommentList);
   }
 
   /**
    * Categorize all score values into different range intervals and count the frequency for each
    * interval, and set the aggregatedValues.
    *
-   * @param sentimentScores a list of score values from -1.0 to 1.0
+   * @param userCommentList a list of userComment analyzed from sentiment analysis with upadted
+   *     score and magnitude
+   * @param lowerEnd lower end boundary of the map
+   * @param upperEnd upper end boundary of the map
+   * @param interval interval for the range
+   * @return a categorized map based on userCommentList from lowerEnd to upperEnd with interval
    */
-  private void setAggregateScores(List<Double> sentimentScores) {
-    aggregateValues = new HashMap<>();
+  private List<SentimentBucket> categorizeToBucketList(
+      List<UserComment> userCommentList,
+      BigDecimal lowerEnd,
+      BigDecimal upperEnd,
+      BigDecimal interval,
```

Comment:
If we are using them as static constants, we don't have to pass them in, unless there is plan to change that behavior?